src/averageAges.js

```diff
@@ -37,16 +47,31 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const allWomen = people.filter(person => person.sex === 'f');
+
+  const age = allWomen.reduce((sum, { died, born }) =>
+    (sum + (died - born)), 0);
+
+  const womenWithChildren = people.filter(person =>
+    person.sex === 'f' && people.find(woman => woman.mother === person.name));
+
+  const difference = womenWithChildren.reduce((sum, { died, born }) =>
+    sum + (died - born), 0);
+
+  return withChildren === true
+    ? difference / womenWithChildren.length
+    : age / allWomen.length;
```

Comment:
the same here. You can check it in the beginning and create the sample based on this arg